src/render/ogc/SDL_render_ogc.c

```diff
@@ -426,8 +427,8 @@ static int OGC_QueueGeometry(SDL_Renderer *renderer, SDL_RenderCommand *cmd, SDL
 static int OGC_RenderSetViewPort(SDL_Renderer *renderer, SDL_RenderCommand *cmd)
 {
     const SDL_Rect *viewport = &cmd->data.viewport.rect;
-
-    OGC_set_viewport(viewport->x, viewport->y, viewport->w, viewport->h);
+    OGC_set_viewport(viewport->x, viewport->y, viewport->w, viewport->h, 
+    viewport->w==640 ? (viewport->h*16.0f/9.0f)+1 : viewport->w);
```

Comment:
Please double check the indentation (in other places, too):
```suggestion
    OGC_set_viewport(viewport->x, viewport->y, viewport->w, viewport->h, 
                     viewport->w==640 ? (viewport->h*16.0f/9.0f)+1 : viewport->w);
```